File: 1546.maximum-number-of-non-overlapping-subarrays-with-sum-equals-target/solution.kt
class Solution {
    fun maxNonOverlapping(nums: IntArray, target: Int): Int {
        var res = 0
        var sum = 0
        val vis = mutableSetOf(0)

        for (num in nums) {
            sum += num
            if (sum - target in vis) {
                res++
                sum = 0
                vis.clear()
            }
            vis += sum
        }

        return res
    }
}

File: 1546.maximum-number-of-non-overlapping-subarrays-with-sum-equals-target/solution.go
func maxNonOverlapping(nums []int, target int) int {
    res := 0
    v := map[int]bool{0: true}
    s := 0

    for _, n := range nums {
        s += n
        r := s-target
        if v[r] {
            res++
            s = 0
            v = make(map[int]bool)
        }
        v[s] = true
    }

    return res
}